File: spring-graphql-querydsl/src/main/java/com/example/demo/repository/PostRepository.java
package com.example.demo.repository;

import com.example.demo.model.Post;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;
import org.springframework.graphql.data.GraphQlRepository;

import java.util.UUID;

@GraphQlRepository()
public interface PostRepository extends JpaRepository<Post, UUID> ,
        QuerydslPredicateExecutor<CommentRepository> {

}

File: dgs-kotlin/src/main/kotlin/com/example/demo/repository/PostRepository.kt
package com.example.demo.repository

import com.example.demo.model.PostEntity
import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.PagingAndSortingRepository
import java.util.*

interface PostRepository : CrudRepository<PostEntity, UUID>,
    PagingAndSortingRepository<PostEntity, UUID> {
    fun findByAuthorId(id: UUID): List<PostEntity>
}